### PR TITLE
Update how softWrap option behaves

### PR DESCRIPTION
**Pull request by marcel0ll**
### Issue or RFC Endorsed by Atom's Maintainers

#11303 

### Description of the Change

Updated how soft wrapping works on atom. Now there are 5 options:

- disabled: Editor wont soft wrap

- window: Editor will wrap at window's width

- preferredLineLength: Editor will wrap at the option  preferred line length
value

- minWindowOrPll: Editor will wrap at the minimum value of window's width
and preferred line length option value

- maxWindowOrPll: Editor will wrap at the maximum value of window's width
and preferred line length option value

This was changed due to the fact that atom would not respect the now removed option "Soft wrap ap preferred line length" when the window was smaller than the preferred line length.

### Alternate Designs

I considered just changing the code that forced atom to wrap lines at the minimum value of preferred line length and window size, but there was already an issue that had a better suggestion than my idea as it was pointed out to me. 

### Possible Drawbacks

Maybe backward compatibility with current atom configuration as the type from softWrap changed from Boolean to String.

### Verification Process

I opened atom in dev mode.
Set soft wrap to 'disabled' and checked the result.
Set soft wrap to 'window' and checked the result.
Set soft wrap to 'preferredLineLength', set preferred line length to 1 and checked the result.
Set soft wrap to 'preferredLineLength', set preferred line length to 100 and checked the result.
Set soft wrap to 'minWindowOrPll', set preferred line length to 1 and checked the result.
Set soft wrap to 'minWindowOrPll', set preferred line length to 100 and checked the result.
Set soft wrap to 'maxWindowOrPll', set preferred line length to 1 and checked the result.
Set soft wrap to 'maxWindowOrPll', set preferred line length to 100 and checked the result.

### Release Notes

- Updated softWrap option to allow more control on how atom wraps the code

### CODE REVIEW

**Comment by rsese** [comment]
Thanks @0tho! Someone from the team will take a look as soon as they can.

**Comment by 0tho** [comment]
@rsese Is there anything I can do to help the review process?

**Comment by rsese** [comment]
Hey @0tho :wave: If anything more is needed from your end, the person that ends up reviewing will comment here so nothing for now.  Your pull request is on the team's list to review as soon as they can, but we can't promise a timeframe for review.

**Comment by 0tho** [comment]
Not sure what the error on appveyor means and how I can fix it. Any help would be appreciated

**Comment by 50Wliu** [comment]
Network error. I restarted the build.

**Comment by rafeca** [comment]
Hey! 👋

We have recently started using [`prettier`](http://prettier.io/) and this has caused major style changes on the Atom JS codebase (you can check [the related PR](https://github.com/atom/atom/pull/19391)).

This is good news: now the Atom code is more consistent and it's much easier to re-format the code to follow the codestyle (now you only need to run `script/lint --fix`).

This change caused conflicts on your PR that we have automatically solved, hope you don't mind 😄

With ❤️, the Atom team.

**Review by ghost — APPROVED** [review]

**Comment by marcel0ll** [comment]
Is there anything else that could be done to merge this PR? Or if this is not something that won't ever be merged for some reason, let's just close it.

It has been 2 years since I opened this PR.

**Comment by michael-camilo** [comment]
Please, someone, merge this PR !!

**Comment by ThatXliner** [comment]
> Please, someone, merge this PR !!

There are conflicting files

**Comment by marcel0ll** [comment]
There weren't for years

**Comment by ThatXliner** [comment]
Welp, there are now

File: src/text-editor.js [hunk 1]
```diff
@@ -4169,20 +4156,33 @@ module.exports = class TextEditor {
...
-      if (this.softWrapAtPreferredLineLength) {
-        return Math.min(this.getEditorWidthInChars(), this.preferredLineLength);
-      } else {
-        return this.getEditorWidthInChars();
+      switch (this.softWrapped) {
+        case 'window':
+          return this.getEditorWidthInChars();
+        case 'preferredLineLength':
+          return this.preferredLineLength;
+        case 'minWindowOrPll':
```
Comment (sadick254) [inline src/text-editor.js:4174]:
```suggestion
        case 'minWindowOrPreferredLineLength':
```
I know this is a long one but it is clear on what the option is supposed to achieve.

**Review by sadick254 — CHANGES_REQUESTED** [review]

**Comment by sadick254** [comment]
Hey @marcel0ll, I know this PR has been open for a long time now (2yrs). Apologies that we didn't get to it on time. Thank you for all the hard work you have done on the PR. However, I don't think this would be merged.

The current options that we have `preferredLineLength` and `window width` wrap are sufficient for soft wrapping. Adding 2 more options to me looks like overkill, it also adds complexity to the wrapping behaviour.

Thank you again for your contribution.

**Comment by marcel0ll** [comment]
Hey @sadick254, no problem. I am honestly glad that this had some closure. Thank you for taking the time to make the previous review and now for closing the PR.

**Pull request closed without merging**